### ISSUE TITLE
Decouple activation to `layer`

### ISSUE THREAD

**Issue by zhoonit** [nnstreamer/nntrainer#153]
Might be related to #152.
### AS-IS
Current implementation of `fc_layer` has activation function and loss function coupled.

### TO-BE
IMO, decoupling the activation layer as well as loss function(as @kparichay noted) is also needed to increase modularity.

for example in Keras, because activation functionality is decoupled, putting `bn_layer` between `fc` and `activation` can be done easily.

```python
model.add(Dense(64))
model.add(BatchNormalization())
model.add(Activation('tanh'))
```

still activation can be used as an argument of `fc_layer` possible because `fc_layer` can have `activation_layer`

```python
model.add(Dense(64, activation="relu" ))
```

### Proposal
- Add `activation_layer`(or equivalent) and extract the functionality.
- Change `fc_layer` to have `activation_layer`

@jijoongmoon Could you confirm if this change is okay?

**Comment by jijoongmoon** [comment]
I thinks it must be done soon. Thanks for addressing this issue.
